### ISSUE TITLE
Warp: How to increase maximum number of files?

### ISSUE THREAD

**Issue by unclechu** [yesodweb/wai#855]
Hey, I’ve implemented a file uploading via multipart/form-data using Warp & Servant but now I’m getting this exception when I’m selecting more than a few files:

```
Maximum number of files exceeded
CallStack (from HasCallStack):
  error, called at ./Network/Wai/Parse.hs:534:29 in wai-extra-3.1.6-22LSJkkgAfcGyKEssmjhv:Network.Wai.Parse
```

How can I increase this maximum number of files for Warp?

**Comment by snoyberg** [comment]
You'd need to change the settings with something like this:

https://www.stackage.org/haddock/lts-18.12/wai-extra-3.1.6/Network-Wai-Parse.html#v:setMaxRequestNumFiles

**Comment by unclechu** [comment]
@snoyberg Yeah, I found this one earlier. But it isn’t obvious for me looking at the Warp’s API how would I use it there.

**Comment by unclechu** [comment]
Ah, I think I should look into the Servant library I’m using:

https://hackage.haskell.org/package/servant-multipart-0.12.1/docs/src/Servant.Multipart.html
